ionoscloud/data_source_autoscaling_group.go

```diff
@@ -0,0 +1,314 @@
+									},
+									"size": {
+										Computed:    true,
+										Type:        schema.TypeInt,
+										Description: "User-defined size for this replica volume in GB.",
+									},
+									"ssh_keys": {
+										Type:     schema.TypeList,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Computed: true,
+									},
+									"type": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: "Storage Type for this replica volume. Possible values: SSD, HDD, SSD_STANDARD or SSD_PREMIUM",
+									},
+									"boot_order": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: `Determines whether the volume will be used as a boot volume: NONE - the volume will not be used as boot volume, PRIMARY - the volume will be used as boot volume, AUTO - will delegate the decision to the provisioning engine to decide whether to use the volume as boot volume.`,
+									},
+									"bus": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: `The bus type of the volume. Default setting is 'VIRTIO'. The bus type 'IDE' is also supported.`,
+									},
+									"backup_unit_id": {
+										Computed:    true,
+										Type:        schema.TypeString,
+										Description: "The uuid of the Backup Unit that user has access to.",
+									},
+								}},
+						},
+					},
+				},
+			},
+			"datacenter_id": {
+				Type:        schema.TypeString,
+				Description: "Unique identifier for the resource",
+				Computed:    true,
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Description: "Location of the data center.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceAutoscalingGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	id, idOk := d.GetOk("id")
+	name, nameOk := d.GetOk("name")
+
+	if idOk && nameOk {
+		return diag.FromErr(fmt.Errorf("id and name cannot be both specified at the same time"))
+	}
+	if !idOk && !nameOk {
+		return diag.FromErr(fmt.Errorf("please provide either the group id or name"))
+	}
+
+	var group autoscaling.Group
+	var err error
+
+	if idOk {
+		group, _, err = client.GetGroup(ctx, id.(string), 2)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occurred while fetching group with ID %s: %w", id.(string), err))
+		}
+	} else {
+		groups, _, err := client.ListGroups(ctx)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occurred while getting groups: %w", err))
+		}
+
+		var results []autoscaling.Group
+
+		if groups.Items != nil {
+			for _, g := range *groups.Items {
+				// TODO: this will not be necessary once the swagger is fixed and list returns the full properties
+				tmpGroup, _, err := client.GetGroup(ctx, id.(string), 2)
```

Comment:
how can `id.(string)` be used here if this is the `else` branch of `if idOk` ?
Won't it be null ?

Reply:
@cbeti-ionos indeed this was a problem, I fixed it before seeing this comment because there were some tests that were failing, this change will be visible in the commit after c9df22e0066ca69a7809de59cd124fe97fec9580, I will make it soon.